analytics/src/main/java/com/segment/analytics/JSMiddleware.java

```diff
@@ -1,18 +1,41 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2014 Segment.io, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
 package com.segment.analytics;
 
 import android.content.Context;
-
-import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
-public class JSMiddleware {
-    protected List<Middleware> sourceMiddleware;
-    protected Map<String, List<Middleware>> destinationMiddleware;
-    protected Context context;
-    protected HashMap<String, Object> settings;
+public abstract class JSMiddleware {
+  protected List<Middleware> sourceMiddleware;
```

Comment:
I'd really like to maintain 4 space tabs

Reply:
oh this is due to the fact that i ran `gradlew spotlessApply`. I think we have it configured to be 2 spaces historically. We can talk abt this change later tho